### ISSUE TITLE
solve dict=True ignored

### ISSUE THREAD

**Issue by rwst** [sympy/sympy#13534]
```
In [22]: solve(Eq(2*x,1), dict=true)
Out[22]: [{x: 1/2}]

In [23]: solve([Eq(x**2-1,0),Gt(x,0)],(x,),dict=True)
Out[23]: x = 1
```

**Comment by AtharvaKhare** [comment]
Can I take this up?

**Comment by rwst** [comment]
Yes, please!

**Comment by czgdp1807** [comment]
Reopening this issue for further discussion. ping @oscarbenjamin @cbm755

**Comment by czgdp1807** [comment]
For the following case, 
```python
>>> solve([x**2 - 1 > 0, x < 10], dict=True)
```
`solve` raises `UserWarning` which from my point of view is okay. However, one way to represent inequalities in the form of `dict` can be is to use `as_set()` and return, `((-oo < x) & (x < -1)) | ((1 < x) & (x < 10))` as `{x: Union(Interval.open(-oo, -1), Interval.open(1, 10))}`. But `as_set()` is a slow method so that can be it's drawback.
For cases like below,
```python
>>> solve([Eq((x-1)*(x-2)*(x-3),0), x>0], dict=True)
Eq(x, 1) | Eq(x, 2) | Eq(x, 3)
```
We can return, `[{x:1}, {x:2}, {x:3}]` or again using `as_set()` will lead to, `{x: FiniteSet(1, 2, 3)}` but again it will slowed down but not much as in the case of inequalities.
What do you say?

**Comment by oscarbenjamin** [comment]
The dict=True flag is documented as
```
        'dict'=True (default is False)
            return list (perhaps empty) of solution mappings
```
When there are inequalities present in the input we can not in general meet that requirement. I don't think it's a good idea to try if we can't. The whole point of the dict=True flag is that the user wants to ensure they get a consistent return type so they don't need to subsequently check the type with isinstance.

**Comment by czgdp1807** [comment]
So probably raising `UserWarning` is the only option we have to make the user aware of the incompatible types.

**Comment by oscarbenjamin** [comment]
If the long term intention is that solve should not accept dict=True in combination with inequalities then a DeprecationWarning would make sense.

Otherwise what really needs to be sorted out is what exactly the API for solve should be. Currently it does a whole range of different things. From memory the docs give many examples but don't clearly define anywhere what should or should not be handled by solve, what types it should return etc. I find it hard to change anything there without a clear documented definition of what solve should do.

Part of the confusion here comes I think from conflicting interpretations of what should happen with inequalities. I think that many users use inequalities in combination with equalities as a way of filtering the solutions to a system of equations. There is though a separate function in solve which attempt to simplify a system of inequalities and return a Boolean as in
```
In [2]: solve([x>1, x+3>2], x)
Out[2]: 1 < x ∧ x < ∞
```
However I think that is usually not what users want solve to do.

Really the issue here is the fact that under certain conditions solve will go off and call `solve_univariate_inequality` which does something very different from the rest of solve and is generally incompatible with dict=True. Perhaps clarifying this in the documentation is a better approach than making any change to the return types.

**Comment by cbm755** [comment]
> I think that many users use inequalities in combination with equalities as a way of filtering the solutions to a system of equations.

Yes this!  Can we try to satisfy such users if possible and raise ValueError (or something) if the results cannot be expressed in a dict?  That seems the most user-friendly to me...

**Comment by cbm755** [comment]
I'm linking #11661 here, which is basically a dupe of this issue.  It contains some test cases that should appear in any tests.
